### ISSUE TITLE
Fail Build On Container Error

### ISSUE THREAD

**Issue by Jotschi** [fabric8io/docker-maven-plugin#488]
Is there a way to let the build fail if the docker command/docker startup fails?

The following example a container is started and the command `ls /bogus` is executed. The command directly fails but the maven build result is success after the timeout is reached. In my usecase I would like to execute a blocking command within the docker container and let the build fail if the command fails.

It seems that the exit code is being ignored?

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
	<modelVersion>4.0.0</modelVersion>

	<groupId>de.jotschi.docker</groupId>
	<version>0.6.14</version>
	<artifactId>tests</artifactId>

	<packaging>pom</packaging>
	<name>Test</name>

	<properties>
		<dockerHostname>plexus</dockerHostname>
		<dockerHostProtocol>tcp</dockerHostProtocol>
	</properties>

	<build>
		<plugins>
			<plugin>
				<groupId>io.fabric8</groupId>
				<artifactId>docker-maven-plugin</artifactId>
				<version>0.15.7</version>
				<executions>
					<execution>
						<id>testing</id>
						<phase>integration-test</phase>
						<goals>
							<goal>start</goal>
						</goals>
					</execution>
				</executions>
				<configuration>
					<dockerHost>${dockerHostProtocol}://${dockerHostname}:2375</dockerHost>
					<verbose>true</verbose>
					<autoPull>always</autoPull>
					<images>
						<image>
							<alias>test</alias>
							<name>jotschi/protractor:latest</name>
							<run>
								<wait>
									<time>30000</time>
								</wait>
								<cmd>
									<shell>ls /bogus</shell>
								</cmd>
								<log>
									<enabled>true</enabled>
									<color>green</color>
								</log>
							</run>
						</image>
					</images>
				</configuration>
			</plugin>
		</plugins>
	</build>
</project>
```

**Comment by rhuss** [comment]
The problem here is that the plugin only starts a container (with `docker:start`) and possibly stops it (with `docker:stop`), but it has no idea when the container stops on its own because of an error (and never evaluates the exit code of the container). Also the container starts up in the background except when you are using `-Ddocker.follow`. Here will block until you hit CTRL-C. It could be a good idea to check the status of the container in that use case and exit Maven when the container exits, too (with the corresponding error code). 

But I guess, this wont help you in your use case, or ?

Another option you could play with today is to wait on a certain log output in your command (for the positive case) with a high enough timeout. Then the build will fail if it hits the timeout, and succeed when it matches the log output. In #252 there is some discussion to enhance the `<wait>` syntax for even more options. Maybe one could introduce a new wait option for waiting the container to finishe and then examine its exit code ? Thats sounds feasible and possibly matches your use case.

**Comment by Jotschi** [comment]
@rhuss I guess the best way to solve this would be an additional wait option. I was just wondering whether I missed a something in the docs. I have already found a workaround in which I invoke a bash script which handles the checks. You can close the issue if you want to. 
Thanks for your feedback.

**Comment by rhuss** [comment]
I've create a new issue #498 for this. Shouldn't be hard to implement but can not promise anything ;-). btw, PR are highly appreciated ;-)
